paasta_tools/cli/cmds/status.py

```diff
@@ -836,232 +801,87 @@ def _print_flink_status_from_job_manager(
-    if flink_jobs.get("jobs"):
-        job_ids = [job.id for job in flink_jobs.get("jobs")]
-    try:
-        jobs = a_sync.block(get_flink_job_details, service, instance, job_ids, client)
-    except Exception as e:
-        output.append(PaastaColors.red("Exception when talking to the API:"))
-        output.append(str(e))
-        return 1
+    # Collect job details
+    job_details = flink_tools.collect_flink_job_details(status, overview, jobs)
 
-    # Avoid cutting job name. As opposed to default hardcoded value of 32, we will use max length of job name
-    if jobs:
-        max_job_name_length = max([len(get_flink_job_name(job)) for job in jobs])
-    else:
-        max_job_name_length = 10
-
-    # Apart from this column total length of one row is around 52 columns, using remaining terminal columns for job name
-    # Note: for terminals smaller than 90 columns the row will overflow in verbose printing
-    allowed_max_job_name_length = min(
-        max(10, shutil.get_terminal_size().columns - 52), max_job_name_length
-    )
+    # Header section
+    output.extend(flink_tools.format_flink_instance_header(instance_details, verbose))
 
-    output.append("    Jobs:")
-    if verbose > 1:
-        output.append(
-            f'      {"Job Name": <{allowed_max_job_name_length}} State       Job ID                           Started'
-        )
-    else:
-        output.append(
-            f'      {"Job Name": <{allowed_max_job_name_length}} State       Started'
+    # Verbose metadata
+    if verbose:
+        output.extend(
+            flink_tools.format_flink_instance_metadata(instance_details, service)
+        )
+        output.extend(
+            flink_tools.format_flink_config_links(service, instance, ecosystem)
+        )
+        output.append(OUTPUT_HORIZONTAL_RULE)
+        output.extend(flink_tools.format_flink_log_commands(service, instance, cluster))
+        output.append(OUTPUT_HORIZONTAL_RULE)
+        output.extend(
+            flink_tools.format_flink_monitoring_links(
+                service, instance, ecosystem, cluster
+            )
         )
+        output.append(OUTPUT_HORIZONTAL_RULE)
 
-    # Use only the most recent jobs
-    unique_jobs = (
-        sorted(jobs, key=lambda j: -j["start_time"])[0]  # type: ignore
-        for _, jobs in groupby(
-            sorted(
-                (j for j in jobs if j.get("name") and j.get("start_time")),
-                key=lambda j: j["name"],
-            ),
-            lambda j: j["name"],
+    # State and pods
+    output.extend(
+        flink_tools.format_flink_state_and_pods(
+            job_details["state"],
+            job_details["pod_counts"],
+            job_details.get("job_counts"),
+            job_details.get("taskmanagers"),
+            job_details.get("slots_available"),
+            job_details.get("slots_total"),
         )
     )
 
-    allowed_max_jobs_printed = 3
-    job_printed_count = 0
+    # Early return if not running
```

Comment:
the previous comment had a little more detail about trying to match kubectl output and handling pods in crashloopbackoff - has that changed?